instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpServerAttributesExtractor.java

```diff
@@ -213,4 +203,42 @@ private String clientIp(REQUEST request) {
   public SpanKey internalGetSpanKey() {
     return SpanKey.HTTP_SERVER;
   }
+
+  private static final class ClientAddressGetter<REQUEST>
+      implements FallbackNamePortGetter<REQUEST> {
+
+    private final HttpServerAttributesGetter<REQUEST, ?> getter;
+
+    private ClientAddressGetter(HttpServerAttributesGetter<REQUEST, ?> getter) {
+      this.getter = getter;
+    }
+
+    @Nullable
+    @Override
+    public String name(REQUEST request) {
+      // try Forwarded
+      String forwarded = firstHeaderValue(getter.getHttpRequestHeader(request, "forwarded"));
+      if (forwarded != null) {
+        forwarded = extractClientIpFromForwardedHeader(forwarded);
+        if (forwarded != null) {
+          return forwarded;
+        }
+      }
+
+      // try X-Forwarded-For
+      forwarded = firstHeaderValue(getter.getHttpRequestHeader(request, "x-forwarded-for"));
+      if (forwarded != null) {
+        return extractClientIpFromForwardedForHeader(forwarded);
+      }
+
+      return null;
+    }
+
+    @Nullable
+    @Override
+    public Integer port(REQUEST request) {
+      // TODO: client.port will be implemented in a future PR
+      return null;
```

Comment:
I'll implement the `client.port` attribute in a separate PR - probably after the release, I think I'll refactor the `FallbackNamePortGetter` util a bit to avoid double parsing (as much as possible)